File: src/intro/the_journey_begins/CheckPalindrome.java
/*
 * Authored by: Jason Wesley Howse
 */

package intro.the_journey_begins;

public class CheckPalindrome {
/*
Given the string, check if it is a palindrome
 */

    boolean checkPalindrome(String inputString) {
        int sLen = inputString.length() - 1;
        for (int i = 0; i < (sLen + 1) / 2; i++) {
            if (inputString.charAt(i) != (inputString.charAt(sLen - i))) {
                return false;
            }//if (inputString.charAt(i) != (inputString.charAt((sLen - 1) - i))) {
        }//for (int i = 0; i < sLen / 2; i++) {
        return true;
    }//boolean checkPalindrome(String inputString) {

/*
Input/Output

[execution time limit] 3 seconds (java)

[input] string inputString

A non-empty string consisting of lowercase characters.

Guaranteed constraints:
1 ≤ inputString.length ≤ 105.

[output] boolean

true if inputString is a palindrome, false otherwise.
 */

}//public class CheckPalindrome {

File: src/_0_intro/_00_the_journey_begins/CheckPalindrome.cs
using System.Linq;

namespace Algorithms.src._0_intro._00_the_journey_begins {
    public class CheckPalindrome {

        /*

Given the string, check if it is a palindrome.

Example

For inputString = "aabaa", the output should be
checkPalindrome(inputString) = true;
For inputString = "abac", the output should be
checkPalindrome(inputString) = false;
For inputString = "a", the output should be
checkPalindrome(inputString) = true.
Input/Output

[execution time limit] 3 seconds (cs)

[input] string inputString

A non-empty string consisting of lowercase characters.

Guaranteed constraints:
1 ≤ inputString.length ≤ 105.

[output] boolean

true if inputString is a palindrome, false otherwise.
        
        */

        public
        bool checkPalindrome(string g) {
            return g.Equals(new string(g.ToCharArray().Reverse().ToArray()));
        }//public bool checkPalindrome(string g) {
    }//public class CheckPalindrome {
}//namespace Algorithms.src._0_intro._00_the_journey_begins {